docs/proposals/hpa/federated-hpa-v2.md

```diff
@@ -0,0 +1,785 @@
+    // +optional
+    // AutoscalingMinThreshold int32                              `json:"autoscalingMinThreshold,omitempty"`
+
+    // StaticWeight is the weight for the target cluster, it only works when the FederatedHPAAssignmentType is StaticWeighted.
+    // +optional
+    StaticWeight            *int32                             `json:"staticWeight,omitempty"`
+    
+    // Priority is the priority for the target cluster, it only works when the FederatedHPAAssignmentType is Prioritized.
+    // +optional
+    Priority                *int32                             `json:"priority,omitempty"`
+}
+
+type FederatedHPAStatus struct {
+	// AggregatedStatus contains the assign results and the status of HPA.
+    AggregatedStatus []autoscalingv2.HorizontalPodAutoscalerStatus `json:"aggregatedStatus,omitempty"`
+}
+```
+
+### Prerequisites
+1. Once the FederatedHPA is applied for the workload, Karmada scheduler should not work for this workload anymore.
+1. Once the FederatedHPA is applied for workload, the replicas changes in member clusters should be retained.
+
+### Initialization HPA assignments
+When FedertedHPA is applied for the first time or updated, the controller will initialize the HPA assignments to member clusters. The assignments should follow the policy configuration. The following contents describe how to assign the HPA to member clusters with different policies.
+
+#### Duplicated assignment policy
+With this policy, FederatedHPA will assign the same minReplicas/maxReplicas(equal to federatedMaxReplicas/federatedMinReplicas) to all member clusters.
+
+So, FederatedHPA controller will create the HPA's works in which the minReplicas/maxReplicas are equal to FederatedMaxReplicas/FederatedMinReplicas in all member clusters.
+And also, FederatedHPA controller will update the ResourceBinding to make all the replicas not less than minReplicas.
+
+Suppose we have the following configuration:
+```
+# FederatedHPA Configuration
+ClusterAffinity:
+  clusterNames:
+    - member1
+    - member2
+    - member3
+    - member5
+FederatedHPAMinReplicas: 3
+FederatedHPAMaxReplicas: 10
+---
+# ResourceBinding
+clusters:
+  - name: member1
+    replicas: 1
+  - name: member2
+    replicas: 4
+  - name: member3
+    replicas: 20
+  - name: member4
+    replicas: 5
+```
+
+After the assignment, the result will be:
+```
+# member 1/2/3 HPA
+minReplicas: 3
+maxReplicas: 10
+# ResourceBinding
+clusters:
+  - name: member1
+    replicas: 3
+  - name: member2
+    replicas: 4 # To avoid overloading, the replicas will not be changed.
+  - name: member3
+    replicas: 10
+  - name: member5
+    replicas: 3
+```
+
+We can see member4 is deleted from ResourceBinding, which means once the FederatedHPA is applied, the selection result of Karmada scheduler will be ignored.
```

Comment:
If `member5` is filtered out because it does not meet the scheduling conditions, but you can add `member5` directly here, is that expected?

Reply:
Yes, it's expected. Once HPA is applied, the selection of Karmada scheduler means nothing.

Reply:
> Once HPA is applied, the selection of Karmada scheduler means nothing.

If the scheduler schedules workloads to `member1`, `member2` and `member3`, and `FederatedHPA` has only `member3` set, and suddenly `member3` fails, what will happen to the workloads at that time?